lib/Dialect/Calyx/Transforms/CalyxHelpers.cpp

```diff
@@ -36,6 +36,22 @@ hw::ConstantOp createConstant(Location loc, OpBuilder &builder,
                                         APInt(width, value, /*unsigned=*/true));
 }
 
+calyx::InstanceOp createInstance(Location loc, OpBuilder &builder,
+                                 ComponentOp component,
+                                 SmallVectorImpl<Type> &resultTypes,
+                                 std::string &instanceName,
+                                 StringRef componentName) {
+  OpBuilder::InsertionGuard g(builder);
+  builder.setInsertionPointToStart(component.getBodyBlock());
+  return builder.create<InstanceOp>(loc, resultTypes, instanceName,
+                                    componentName);
+}
+
+std::string getInstanceName(mlir::func::CallOp callOp) {
+  SmallVector<StringRef, 2> strVet = {callOp.getCallee(), "instance"};
```

Comment:
I think you could also do:

```cpp
return llvm::join_items(/separator=*/"_", callOp.getCallee(), "instance");
```

or

```cpp
return llvm::join({callOp.getCallee(), "instance"}, /separator=*/"_");
```

Reply:
The use of SmallVector here is necessary, otherwise it won't compile.